### PR TITLE
API Issues: 307 & 372

### PR DESCRIPTION
**Pull request by venkatvghub**
@StSarc , please review.

This is a fix for https://github.com/razorpay/api/issues/372 and https://github.com/razorpay/api/issues/307.

The necessary client code for refactoring this is available in https://github.com/razorpay/api/tree/f/307. Sending you a separate PR for that.

Most code follows our discussion we had.

### CODE REVIEW

File: src/config/config.php [hunk 1]
```diff
@@ -119,4 +119,7 @@
   // Allow Markdown in all fields
   // 'markdown_in_attachments' => ['pretext', 'text', 'title', 'fields', 'fallback']
 
+  // Team details
+  'team' => 'razorpay',
```
Comment (StSarc) [inline src/config/config.php]: Doesn't seem to be being used anywhere. We can remove this.

**Comment by venkatvghub** [comment]
@StSarc , checked with redis - success and failure with the jobs getting deleted, both on queue and send. Please review. No changes on the api branch: https://github.com/razorpay/api/pull/396

**Comment by venkatvghub** [comment]
@shashankkumar, please review these changes.

File: src/Client.php [hunk 2]
```diff
@@ -112,12 +137,16 @@ public function __construct($endpoint, array $attributes = [], Queue $queue = nu
 
     if (isset($attributes['markdown_in_attachments'])) $this->setMarkdownInAttachments($attributes['markdown_in_attachments']);
 
+    if(isset($attributes['is_slack_enabled'])) $this->setSlackStatus($attributes['is_slack_enabled']);
```
Comment (StSarc) [inline src/Client.php]: Let this be a block?

Comment (shashankkumar) [inline src/Client.php]: Yeah, the previous line too.

**Pull request closed without merging**